src/WebJobs.Script/Description/Worker/WorkerLanguageInvoker.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Azure.WebJobs.Script.Binding;
+using Microsoft.Azure.WebJobs.Script.Description.Script;
+using Microsoft.CodeAnalysis;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    // TODO: make this internal
+    public class WorkerLanguageInvoker : FunctionInvokerBase
+    {
+        private readonly Collection<FunctionBinding> _inputBindings;
+        private readonly Collection<FunctionBinding> _outputBindings;
+        private readonly BindingMetadata _trigger;
+        private readonly Action<ScriptInvocationResult> _handleScriptReturnValue;
+        private readonly BufferBlock<ScriptInvocationContext> _invocationBuffer;
+
+        internal WorkerLanguageInvoker(ScriptHost host, BindingMetadata trigger, FunctionMetadata functionMetadata,
```

Comment:
We have test classes for all our other invoker types. Recommend adding one for this. For now it might just have tests for the few private helpers you have.